kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/players/BaseAudioPlayer.kt

```diff
@@ -143,15 +145,48 @@ abstract class BaseAudioPlayer internal constructor(private val context: Context
         exoPlayer = ExoPlayer.Builder(context).apply {
             if (bufferConfig != null) setLoadControl(setupBuffer(bufferConfig))
         }.build()
+        forwardingPlayer = createForwardingPlayer()
 
         mediaSession.isActive = true
 
-        notificationManager = NotificationManager(context, exoPlayer, mediaSession.sessionToken, notificationEventHolder)
+        val playerToUse = if (playerConfig.interceptPlayerActionsTriggeredExternally) forwardingPlayer else exoPlayer
+
+        notificationManager = NotificationManager(context, playerToUse, mediaSession.sessionToken, notificationEventHolder)
 
-        mediaSession.setCallback(MediaSessionListener())
         exoPlayer.addListener(PlayerListener())
-        exoPlayer.stop()
-        mediaSessionConnector.setPlayer(exoPlayer)
+        mediaSessionConnector.setPlayer(playerToUse)
+    }
+
+    private fun createForwardingPlayer(): ForwardingPlayer {
+        return object : ForwardingPlayer(exoPlayer) {
+            override fun play() {
+                playerEventHolder.updateOnMediaSessionCallbackTriggered(MediaSessionCallback.PLAY)
```

Comment:
If `ForwardingPlayer` is used, the player actions will trigger our custom player events.